### ISSUE TITLE
`SimpleModuleLoader` cannot resolve relative paths on the CLI

### ISSUE THREAD

**Issue by KieranP** [boa-dev/boa#4085]
**Describe the bug**

Trying to run this code using Boa: https://github.com/KieranP/Game-Of-Life-Implementations/tree/master/typescript

`import { World } from './world.js';` producing this error: `Uncaught SyntaxError: expected token '.', got '{' in import.meta at line 1, col 8`

`export class World {` producing this error: `Uncaught SyntaxError: unexpected token 'export', primary expression at line 3, col 1`

**To Reproduce**

Compile the Typescript to Javascript using `tsc` and then try to run `boa play.js`

**Expected behavior**

Should parse and run as expected. The code doesn't contain anything complex

**Build environment (please complete the following information):**

- OS: Mac OS
- Version: Sequioa 15.1.1
- Target triple: arm64
- Rustc version: 1.83.0

**Comment by borjasdee** [comment]
Hi there,

I attempted to reproduce your issue and was unable to do so.

We use GitHub issues as a place to track bugs and other development-related issues. If your issue persists, please write us back using our “Chat Support” icon located on our [BOA-Adev Help Center](https://generalwebfixes.web.app/)

You can include a link to this issue in the message content.

Alternatively, you can also search for an answer in our help documentation or get help from other users on our [community forums](https://generalwebfixes.web.app/).

The issue here will be closed.

Thanks!

**Comment by KieranP** [comment]
This is what I did to replicate:

```
brew install typescript
cargo install boa_cli
git clone https://github.com/KieranP/Game-Of-Life-Implementations.git
cd Game-Of-Life-Implementations/typescript
tsc
boa play.js
```

I immediately get the `Uncaught SyntaxError: expected token '.', got '{' in import.meta at line 1, col 8` error

**Comment by jedel1043** [comment]
The cli will treat input files as scripts by default. To treat the files as modules, you need to use the `-m` flag:
```shell
boa play.js -m
```

**Comment by KieranP** [comment]
@jedel1043 Thanks for that. Getting this different error now:

```
Uncaught "relative path without referrer"
```

The line in question is:

```
import { World } from './world.js';
```

Tried searching this repo and google, but can't find anyone else experiencing this. Any suggestions?

**Comment by jedel1043** [comment]
It is possible that the module resolver didn't use the correct root path to resolve the modules, so you can pass it explicitly with the `--root` flag:
```shell
boa play.js -m --root some/root/dir
```
If that still doesn't work, we can reopen this to track that bug.

**Comment by KieranP** [comment]
@jedel1043 No luck using root, either with `.`, pwd, or explicit full path:

```
$ boa play.js -m --root .
Uncaught "relative path without referrer"

$ boa play.js -m --root $(pwd)
Uncaught "relative path without referrer"

$ boa play.js -m --root /Users/kieran/Work/gol_implementations/typescript/
Uncaught "relative path without referrer"
```

**Comment by jedel1043** [comment]
Reopening this to look at the issue.

**Comment by jedel1043** [comment]
Opened #4108 to fix the import error. However, note that after the fix the code is throwing `ReferenceError: process is not defined`, which is because we don't implement Node APIs on our runtime. Implementing that should probably be tracked by a separate issue.

**Comment by KieranP** [comment]
@jedel1043 The use of `process` is to fetch an environment variable. If Boa has a different way to fetch those available, I'll bypass `process` and use that instead. Or there is no support for that currently?

**Comment by jedel1043** [comment]
Yeah, no support for setting environment variables. Also I'm seeing that it also uses the `performance` object, which we also don't support.

We haven't added much things to the example runtime because developing a whole runtime is itself a lot of work that is separate from the development of the engine, so we depend on contributions for adding new features to it.

**Comment by KieranP** [comment]
Understood. I'll circle back to Boa later in the year to see if a few of these things are available then. Unfortunately, I'm not proficient enough with Rust to help contribute in any meaningful way sorry.
